python/graphframes/tests.py

```diff
@@ -15,63 +15,72 @@
 # limitations under the License.
 #
 
-import sys
+import os
 import tempfile
 import shutil
 import re
 
-if sys.version_info[:2] <= (2, 6):
-    try:
-        import unittest2 as unittest
-    except ImportError:
-        sys.stderr.write('Please install unittest2 to test with Python 2.6 or earlier')
-        sys.exit(1)
-else:
-    import unittest
-
-from pyspark import SparkContext
-from pyspark.sql import functions as sqlfunctions, SparkSession
+import pytest
+from pyspark import SparkConf, SparkContext
+from pyspark.sql import functions as F, SparkSession
 
 from .graphframe import GraphFrame, Pregel, _java_api, _from_java_gf
 from .lib import AggregateMessages as AM
 from .examples import Graphs, BeliefPropagation
 
+
+VERSION = open("version.sbt").read().strip()
+
+
+@pytest.fixture(scope="class", autouse=True)
+def set_spark(request, spark_session):
+    request.cls.spark = spark_session
+
+
+@pytest.mark.usefixtures("set_spark")
 class GraphFrameTestUtils(object):
 
     @classmethod
     def parse_spark_version(cls, version_str):
-        """ take an input version string
-            return version items in a dictionary
+        """take an input version string
+        return version items in a dictionary
         """
-        _sc_ver_patt = r'(\d+)\.(\d+)(\.(\d+)(-(.+))?)?'
+        _sc_ver_patt = r"(\d+)\.(\d+)(\.(\d+)(-(.+))?)?"
         m = re.match(_sc_ver_patt, version_str)
         if not m:
-            raise TypeError("version {} shoud be in <major>.<minor>.<maintenance>".format(version_str))
+            raise TypeError(
+                "version {} shoud be in <major>.<minor>.<maintenance>".format(version_str)
+            )
         version_info = {}
         try:
-            version_info['major'] = int(m.group(1))
+            version_info["major"] = int(m.group(1))
         except:
             raise TypeError("invalid minor version")
         try:
-            version_info['minor'] = int(m.group(2))
+            version_info["minor"] = int(m.group(2))
         except:
             raise TypeError("invalid major version")
         try:
-            version_info['maintenance'] = int(m.group(4))
+            version_info["maintenance"] = int(m.group(4))
         except:
-            version_info['maintenance'] = 0
+            version_info["maintenance"] = 0
         try:
-            version_info['special'] = m.group(6)
+            version_info["special"] = m.group(6)
         except:
             pass
         return version_info
 
     @classmethod
     def createSparkContext(cls):
-        cls.sc = sc = SparkContext('local[4]', "GraphFramesTests")
+        cls.conf = SparkConf().setAppName("GraphFramesTests")
+        cls.conf.set(
+            "spark.submit.pyFiles",
+            os.path.abspath("python/dist/graphframes-{VERSION}-py3-none-any.whl"),
+        )
+        cls.sc = SparkContext(master="local[4]", appName="GraphFramesTests", conf=cls.conf)
```

Comment:
Could we avoid it at all? The problem is that Spark Connect does not support SparkContext at all. And I don't see actual usage of `cls.sc`. Can we rely on the `SparkSession` only? As I remember it is also recommended by PySpark documentation too.